### ISSUE TITLE
Error while processing request: (void-variable translate-url)

### ISSUE THREAD

**Issue by orgtre** [gkowzan/biblio-zotero#1]
Thanks for this great biblio add-on! I can get everything to run and it is very useful (I almost don't need Zotero itself anymore now).

However, I get the following error when trying to run `biblio-zotero-insert-bibtex` directly after Emacs starts: `Error while processing request: (void-variable translate-url)`. To check whether this has to do with the order in which packages are loaded, I then tried to `require`/`load` biblio-zotero  (also with exact absolute file path), but now I got: `Error while processing request: (void-variable callback)`.

Only after I visit biblio-zotero.el and run `eval-buffer` everything works. This is on Emacs 29.0.50. Do you have any idea what could cause this behavior or how I could avoid it?

**Comment by gkowzan** [comment]
Do you know if this problem occurs for stable Emacs version, 28.1? If it's only for 29.0.50, could you give me the specific version: snapshot date, commit hash or were did you download it from?

**Comment by orgtre** [comment]
Yes, I get the same behavior on the version 28.1. I'm on macOS and have tried with minimal init.el and fresh installs from the following sources: 
`brew install emacs`
`brew install --cask emacs`
`brew install emacs-plus@29`
The behavior is the same in all cases. 

A workaround is to put the following in init.el:
```
(find-file "path-to/biblio-zotero.el")
(eval-region (point-min) (point-max))
(kill-current-buffer))
```

But if I replace the eval-region call with any of the following it doesn't work anymore:
`(eval-buffer)`
or
`(call-interactively 'eval-buffer)`
or
`(mark-whole-buffer)`
`(call-interactively 'eval-region)`

This also doesn't work:
```
(with-current-buffer
     (find-file-noselect
     "path-to/biblio-zotero.el"
     (eval-region (point-min) (point-max))))
```

**Comment by gkowzan** [comment]
Thanks, could you set `biblio-synchronous` to `t`, `toggle-debug-on-error`, try again and paste the debugging output? Cold you also check if `doi-insert` from biblio works for you?

**Comment by orgtre** [comment]
When setting `biblio-synchronous` to `t`, toggling `toggle-debug-on-error` on directly after startup, and then running `biblio-zotero-insert-bibtex`, I get no traceback but only the message: `Error while processing request: (wrong-number-of-arguments (1 . 1) 0)`. Like before, after visiting biblio-zotero.el and running `eval-buffer` it works.

I run the latest version of [biblio.el](https://github.com/cpitclaudel/biblio.el), which doesn't include a function named exactly `doi-insert`, but `biblio-doi-insert-bibtex` works fine, also directly after startup.
